pkg/codegen/go/gen_program_expressions.go

```diff
@@ -382,13 +423,14 @@ func (g *generator) genScopeTraversalExpression(w io.Writer, expr *model.ScopeTr
 		case "value":
 			g.Fgenf(w, "val0")
 		case "key":
-			g.Fgenf(w, "key0")
+			g.Fgenf(w, "i0")
 		default:
 			contract.Failf("unexpected traversal on range expression: %s", part)
 		}
 	} else {
 		g.Fgen(w, rootName)
-		g.genRelativeTraversal(w, expr.Traversal.SimpleSplit().Rel, expr.Parts, objType)
+		isRootResource := false
+		g.genRelativeTraversal(w, expr.Traversal.SimpleSplit().Rel, expr.Parts[1:], isRootResource)
```

Comment:
nit: I've usually seen this written in the following form elsewhere in the pulumi/pulumi code:
```suggestion
		g.genRelativeTraversal(w, expr.Traversal.SimpleSplit().Rel, expr.Parts[1:], false /* isRootResource */)
```

Reply:
Ah, I will keep this in mind for my future work in pulumi/pulumi. Unfortunately, I've used this pattern all over the go program gen (especially for isInput), so I'll just leave this as-is for consistency within this package.